### ISSUE TITLE
Update of Packages and system failing.

### ISSUE THREAD

**Issue by cgcomer** [ublue-os/bazzite#1239]
### Describe the bug

First attempted to update 4 packages requiring updates via Discover.
![image](https://github.com/ublue-os/bazzite/assets/112912804/524f23f8-4793-4413-b140-d2c37ed00029)

All 4 updates failed with the following error codes.

`Aborted due to failure (While pulling runtime/org.freedesktop.Platform.GL.default/x86_64/22.08 from remote flathub: opcode close: Corrupted file object; checksum expected='ce70fd8dc1e4846798673264ca1af30c33410d54046c05e6a5e30c4424cc416c' actual='ac1d008566082267a54a728ed35e6292fd5de61df22fd69cdaa5c6e0b4e0b058')`

`Aborted due to failure (While pulling runtime/org.freedesktop.Platform.GL.default/x86_64/22.08-extra from remote flathub: opcode close: Corrupted file object; checksum expected='ce70fd8dc1e4846798673264ca1af30c33410d54046c05e6a5e30c4424cc416c' actual='ac1d008566082267a54a728ed35e6292fd5de61df22fd69cdaa5c6e0b4e0b058')`

Attempted update via command line with ujust update. Received the following error:

```
── 22:56:40 - System update ────────────────────────────────────────────────────
Pulling manifest: ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
Checking out tree 45ad5ad... done
Enabled rpm-md repositories: copr:copr.fedorainfracloud.org:matte-schwartz:sunshine copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager copr:copr.fedorainfracloud.org:rodoma92:rmlint copr:copr.fedorainfracloud.org:rok:cdemu fedora-cisco-openh264 updates fedora rpmfusion-free-updates-testing rpmfusion-free-updates rpmfusion-free hardware_razer updates-archive
Updating metadata for 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:rmlint'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rok:cdemu'... done
Updating metadata for 'updates'... done
Updating metadata for 'rpmfusion-free-updates-testing'... done
Updating metadata for 'rpmfusion-free-updates'... done
Updating metadata for 'hardware_razer'... done
Updating metadata for 'updates-archive'... done
Importing rpm-md... done
rpm-md repo 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine'; generated: 2024-06-02T08:36:27Z solvables: 2
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager'; generated: 2024-04-20T12:39:00Z solvables: 12
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:rmlint'; generated: 2024-05-10T10:16:28Z solvables: 4
rpm-md repo 'copr:copr.fedorainfracloud.org:rok:cdemu'; generated: 2024-02-20T12:07:53Z solvables: 23
rpm-md repo 'fedora-cisco-openh264' (cached); generated: 2024-03-12T11:45:42Z solvables: 3
rpm-md repo 'updates'; generated: 2024-06-17T01:23:05Z solvables: 18048
rpm-md repo 'fedora' (cached); generated: 2024-04-14T18:51:11Z solvables: 74881
rpm-md repo 'rpmfusion-free-updates-testing'; generated: 2024-06-14T06:38:19Z solvables: 13
rpm-md repo 'rpmfusion-free-updates'; generated: 2024-06-14T06:37:04Z solvables: 127
rpm-md repo 'rpmfusion-free' (cached); generated: 2024-04-20T12:11:51Z solvables: 422
rpm-md repo 'hardware_razer'; generated: 2024-05-13T15:52:54Z solvables: 15
rpm-md repo 'updates-archive'; generated: 2024-05-22T01:41:39Z solvables: 13161
Resolving dependencies... done
Checking out packages... done
Running pre scripts... done
Running post scripts... done
Running posttrans scripts... done
Writing rpmdb... done
Generating initramfs... done
Writing OSTree commit... done
error: Committing: Adding composefs metadata: Checking out composefs: Processing 8d935243319cb658d019f9605d9b7cb7c5e684fb69673f6454b8241e890d92a3: No data available
System update failed:
   0: Command failed: `/usr/bin/rpm-ostree upgrade`
   1: `/usr/bin/rpm-ostree` failed: exit status: 1

Location:
   src/steps/os/linux.rs:273
Retry? (y)es/(N)o/(s)hell/(q)uiterror: Recipe `update` failed with exit code 1
```

### What did you expect to happen?

Expected either the Discover app to successfully update the packages in question or for the ujust update command to do so.

### Output of `rpm-ostree status`

```shell
State: idle
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
                   Digest: sha256:99cce7eb105cbdc027f1c72399a9ef88458818eeae7a172bbbdc8bb14d2bb22a
                  Version: 40.20240611.0 (2024-06-11T17:00:05Z)
          LayeredPackages: edk2-ovmf qemu virt-manager
            LocalPackages: lact-0.5.4-0.x86_64
                Initramfs: regenerate

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
                   Digest: sha256:99cce7eb105cbdc027f1c72399a9ef88458818eeae7a172bbbdc8bb14d2bb22a
                  Version: 40.20240611.0 (2024-06-11T17:00:05Z)
          LayeredPackages: edk2-ovmf qemu virt-manager
            LocalPackages: lact-0.5.4-0.x86_64
```

### Hardware

Operating System: Bazzite 40
KDE Plasma Version: 6.0.5
KDE Frameworks Version: 6.3.0
Qt Version: 6.7.1
Kernel Version: 6.8.12-301.fsync.fc40.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 24 × AMD Ryzen 9 3900X 12-Core Processor
Memory: 62.7 GiB of RAM
Graphics Processor: AMD Radeon RX 6900 XT
Manufacturer: ASUS
Asus Crosshair Hero VIII Wifi
Installed Main Drive: Samsung SSD 970 EVO Plus 2TB NVME (no SMART errors)
BIOS: 4805
Secure Boot: On (despite what Info Center claims)

**Comment by KyleGospo** [comment]
You have two separate issues here, one is a corrupt file when you do rpm-ostree update, another is a corrupt file when you do a flatpak update.

This could be a network issue, a memory issue, or a drive issue. You might try doing a flatpak repair and clearing the cache via rpm-ostree. If these issues come back you have something more serious going on.

**Comment by cgcomer** [comment]
Okay so here is what I have tried thus far.
Started with `sudo flatpak repair`.  Got the following output

```
Working on the system installation at /var/lib/flatpak
[1/103] Verifying flathub:runtime/org.freedesktop.Platform.ffmpeg_full.i386/x86_[2/103] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.vkBasalt/[17/103] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.MangoHud[22/103] Verifying flathub:runtime/io.github.fastrizwaan.WineZGUI.Locale/x86_64/[25/103] Verifying flathub:runtime/com.obsproject.Studio.Plugin.OBSVkCapture/x86[31/103] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.OBSVkCap[33/103] Verifying flathub:runtime/io.github.dvlv.boxbuddyrs.Locale/x86_64/stabl[42/103] Verifying flathub:app[43/103] Verifying flathub:app/org.[51/103] V[94/103] Verifying fedora:runtime/org.fedoraproject.KDE6Platform/x86_64/f40…
fsck content object a118bf1a57e571d73f37c61fa88b6039d284f5a9ded7e86026e87e198d3b9f47: Corrupted file object; checksum expected='a118bf1a57e571d73f37c61fa88b6039d284f5a9ded7e86026e87e198d3b9f47' actual='ee2d432669f7a190e4d05ff0bf0d3a932eb4c3dcf18a112ed79cfb6911bc72e6', deleting object
fsck content object ddbfa1265b8598e96dbdd0cfd7a9db2ccce4fc7f2fc3b51535ec3043b4f39b49: Corrupted file object; checksum expected='ddbfa1265b8598e96dbdd0cfd7a9db2ccce4fc7f2fc3b51535ec3043b4f39b49' actual='ef47b0ef546becaaa8658ed4e418a30749b1221a94dcf94f31f2b7ddf1c0083c', deleting object
Marking commit as partial: 18a057dc38b0668a51a45743609dacb857058a9a65eee631c5896a1281fc1451
Deleting ref fedora:runtime/org.fedoraproject.KDE6Platform/x86_64/f40 due to invalid objects
[103/103] Verifying flathub:app/com.github.tchx84.Flatseal/x86_64/stable…
Checking remotes...
Pruning objects
Erasing .removed
Reinstalling removed refs
Installing runtime/org.fedoraproject.KDE6Platform/x86_64/f40
Error: Failed to install org.fedoraproject.KDE6Platform: Checksum digest did not match (sha256:0d91586a50f9df3d743365e3f47555b6a8814ad3dc459f71f7fadffd8fd32542 != 18dd1ec15b24645d2613c83b055afc47f59511a9edeb1f3fd85b6c184dfeb008)
```
So I tried `rpm-ostree cleanup -m`, which didn't appear to do anything or at the very least I didn't get any output.
Tried `flatpak repair` again with the same results as before.

After doing some additional research, I tried `rpm-ostree cleanup -bm`. Which failed with some output that instructed to run `journalctl -xe` to see the error log (see attached journal output.txt)
[journal output.txt](https://github.com/user-attachments/files/15879447/journal.output.txt).  

At this point while copying the output from that command to a text file to attach; the terminal crashed so lost the initial output from the cleanup command. However I ran `rpm-ostree cleanup -bm` again on relaunching the Terminal (Ptyxis) expecting the same result however I got a completely different result :
```
ccomer@main-bazzite:~$ rpm-ostree cleanup -bm
Freed: 2.6 GB (pkgcache branches: 0)
```
I thought okay maybe it just needed some time and tried `flatpak repair` again:

```
ccomer@main-bazzite:~$ flatpak repair
Working on the system installation at /var/lib/flatpak
Privileges are required to make changes; assuming --dry-run
[1/98] Verifying flathub:runtime/org.freedesktop.Platform.ffmpeg_full.i386/x86_6[2/98] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.vkBasalt/x[17/98] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.MangoHud/[22/98] Verifying flathub:runtime/io.github.fastrizwaan.WineZGUI.Locale/x86_64/s[24/98] Verifying flathub:runtime/com.obsproject.Studio.Plugin.OBSVkCapture/x86_[30/98] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.OBSVkCapt[32/98] Verifying flathub:runtime/io.github.dvlv.boxbuddyrs.Locale/x86_64/stable[43/98] Verifying flathub:runtime/com.obsproject.Studio.Plugin.GStreamerVaapi/x8[49/98] Verifying flathub:runtime/org.freedesktop.Platform.Compat.i386/x86_64/23[52/98] Verifying flathub:runtime/org.freedesktop.Platform.GL32.default/x86_64/2[59/98] Verifying flathub:runtime/org.freedesktop.Platform.openh264/x86_64/2.4.1[62/98] Verifying flathub:runtime/org.freedesktop.Platform.GL.default/x86_64/23.[69/98] Verifying flathub:runtime/io.github.flattool.Warehouse.Locale/x86_64/sta[82/98] Verifying flathub:runtime/org.freedesktop.Platform.GL.default/x86_64/23.[90/98] Verifying flathub:runtime/com.obsproject.Studio.Plugin.Gstreamer/x86_64/[94/98] Verifying flathub:runtime/org.freedesktop.Platform.ffmpeg-full/x86_64/23[98/98] Verifying flathub:app/com.github.tchx84.Flatseal/x86_64/stable…
Checking remotes...
```

Looks good, I guess, let's try `ujust update` again:

ccomer@main-bazzite:~$ ujust update

```
── 21:00:32 - System update ────────────────────────────────────────────────────
Pulling manifest: ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
Checking out tree 45ad5ad... done
Enabled rpm-md repositories: copr:copr.fedorainfracloud.org:matte-schwartz:sunshine copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager copr:copr.fedorainfracloud.org:rodoma92:rmlint copr:copr.fedorainfracloud.org:rok:cdemu fedora-cisco-openh264 updates fedora rpmfusion-free-updates-testing rpmfusion-free-updates rpmfusion-free hardware_razer updates-archive
Updating metadata for 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:rmlint'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rok:cdemu'... done
Updating metadata for 'fedora-cisco-openh264'... done
Updating metadata for 'updates'... done
Updating metadata for 'fedora'... done
Updating metadata for 'rpmfusion-free-updates-testing'... done
Updating metadata for 'rpmfusion-free-updates'... done
Updating metadata for 'rpmfusion-free'... done
Updating metadata for 'hardware_razer'... done
Updating metadata for 'updates-archive'... done
Importing rpm-md... done
rpm-md repo 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine'; generated: 2024-06-02T08:36:27Z solvables: 2
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager'; generated: 2024-04-20T12:39:00Z solvables: 12
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:rmlint'; generated: 2024-05-10T10:16:28Z solvables: 4
rpm-md repo 'copr:copr.fedorainfracloud.org:rok:cdemu'; generated: 2024-02-20T12:07:53Z solvables: 23
rpm-md repo 'fedora-cisco-openh264'; generated: 2024-03-12T11:45:42Z solvables: 3
rpm-md repo 'updates'; generated: 2024-06-17T01:23:05Z solvables: 18048
rpm-md repo 'fedora'; generated: 2024-04-14T18:51:11Z solvables: 74881
rpm-md repo 'rpmfusion-free-updates-testing'; generated: 2024-06-14T06:38:19Z solvables: 13
rpm-md repo 'rpmfusion-free-updates'; generated: 2024-06-14T06:37:04Z solvables: 127
rpm-md repo 'rpmfusion-free'; generated: 2024-04-20T12:11:51Z solvables: 422
rpm-md repo 'hardware_razer'; generated: 2024-05-13T15:52:54Z solvables: 15
rpm-md repo 'updates-archive'; generated: 2024-05-22T01:41:39Z solvables: 13161
Resolving dependencies... done
Will download: 1 package (86.4 kB)
Downloading from 'updates'... done
Importing packages... done
Checking out packages... done
Running pre scripts... done
Running post scripts... done
Running posttrans scripts... done
Writing rpmdb... done
Generating initramfs... done
Writing OSTree commit... done
error: Committing: Adding composefs metadata: Checking out composefs: Processing 8d935243319cb658d019f9605d9b7cb7c5e684fb69673f6454b8241e890d92a3: No data available
System update failed: 
   0: Command failed: `/usr/bin/rpm-ostree upgrade`
   1: `/usr/bin/rpm-ostree` failed: exit status: 1

Location:
   src/steps/os/linux.rs:273
Retry? (y)es/(N)o/(s)hell/(q)uiterror: Recipe `update` failed with exit code 1
```
At this point, I'm at a loss, I'm going to reboot attempt everything again to see if I get different results.

**Comment by cgcomer** [comment]
It appears that sudo flatpak repair worked this time.

```
ccomer@main-bazzite:~$ flatpak repair
Working on the system installation at /var/lib/flatpak
Privileges are required to make changes; assuming --dry-run
[1/100] Verifying flathub:runtime/org.freedesktop.Platform.ffmpeg_full.i386/x86_[3/100] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.vkBasalt/[18/100] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.MangoHud[23/100] Verifying flathub:runtime/io.github.fastrizwaan.WineZGUI.Locale/x86_64/[31/100] Verifying flathub:runtime/org.freedesktop.Platform.VulkanLayer.OBSVkCap[33/100] Verifying flathub:runtime/io.github.dvlv.boxbuddyrs.Locale/x86_64/stabl[37/100] Verifying flathub:runtime/com.obsproject.Studio.Plugin.OBSVkCapture/x86[45/100] Verifying flathub:runtime/com.obsproject.Studio.Plugin.GStreamerVaapi/x[51/100] Verifying flathub:runtime/org.freedesktop.Platform.Compat.i386/x86_64/2[54/100] Verifying flathub:runtime/org.freedesktop.Platform.GL32.default/x86_64/[61/100] Verifying flathub:runtime/org.freedesktop.Platform.openh264/x86_64/2.4.[64/100] Verifying flathub:runtime/org.freedesktop.Platform.GL.default/x86_64/23[68/100] Verifying flathub:app/net.davidotek.pup[75/100] Verifying fedora:app/org.kde.okular/x86_64/st[100/100] Verifying flathub:app/com.github.tchx84.Flatseal/x86_64/stable…
Checking remotes...
ccomer@main-bazzite:~$ sudo flatpak repair
[sudo] password for ccomer: 
Working on the system installation at /var/lib/flatpak
[100/100] Verifying flathub:app/com.github.tchx84.Flatseal/x86_64/stable…
Checking remotes...
Pruning objects
Reinstalling removed refs
Installing runtime/org.fedoraproject.KDE6Platform/x86_64/f40
Installing runtime/org.freedesktop.Platform/x86_64/23.08
Installing runtime/org.gnome.Platform/x86_64/45
Installing runtime/org.gnome.Platform/x86_64/46
Installing runtime/org.kde.Platform/x86_64/6.7
```
I'm going to try a second reboot before attempting the update or flatpak update via Discover again just to be safe.

**Comment by cgcomer** [comment]
After some weird reboots and kde have some odd display issues, I rebooted into the 1 option in the Grub Menu.  This got me to what appeared to be a stable image (even though the versions listed on both the 0 and 1 options are the same).

Discover showed the same packages as needing updated.  This time I tried `flatpak repair --user`.  That appeared to update 3 of the packages listed (notably the Mesa packages), leaving Chrome and one other (I don't remember which).  Using Discover's built-in update function appeared to update these.  Lesson to self: don't overthink it.

Now to figure out how to force Bazzite to boot the 1 option from Grub rather than the 0 which appears to have borked KDE.

**Comment by KyleGospo** [comment]
0 is your latest deployment, you can select the previous one with `rpm-ostree rollback`

**Comment by cgcomer** [comment]
Thanks. I figured it out.  
I haven't checked if there are still any corrupt files when running the `rpm-ostree update` yet (it's late).  I'll try that tomorrow with my fingers crossed.

I mainly just wish I knew what is causing this to begin with.

**Comment by cgcomer** [comment]
Okay, I had some time after work.  First, flatpak updates from Discover worked fine.  However, upon attempting a System update to check if it is still having issues:

```
── 20:26:31 - System update ────────────────────────────────────────────────────
Pulling manifest: ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
Checking out tree 1baffac... done
Enabled rpm-md repositories: copr:copr.fedorainfracloud.org:matte-schwartz:sunshine copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager copr:copr.fedorainfracloud.org:rodoma92:rmlint copr:copr.fedorainfracloud.org:rok:cdemu fedora-cisco-openh264 updates fedora rpmfusion-free-updates-testing rpmfusion-free-updates rpmfusion-free hardware_razer updates-archive
Importing rpm-md... done
rpm-md repo 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine' (cached); generated: 2024-06-02T08:36:27Z solvables: 2
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager' (cached); generated: 2024-04-20T12:39:00Z solvables: 12
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:rmlint' (cached); generated: 2024-05-10T10:16:28Z solvables: 4
rpm-md repo 'copr:copr.fedorainfracloud.org:rok:cdemu' (cached); generated: 2024-02-20T12:07:53Z solvables: 23
rpm-md repo 'fedora-cisco-openh264' (cached); generated: 2024-03-12T11:45:42Z solvables: 3
rpm-md repo 'updates' (cached); generated: 2024-06-18T10:02:39Z solvables: 18127
rpm-md repo 'fedora' (cached); generated: 2024-04-14T18:51:11Z solvables: 74881
rpm-md repo 'rpmfusion-free-updates-testing' (cached); generated: 2024-06-14T06:38:19Z solvables: 13
rpm-md repo 'rpmfusion-free-updates' (cached); generated: 2024-06-14T06:37:04Z solvables: 127
rpm-md repo 'rpmfusion-free' (cached); generated: 2024-04-20T12:11:51Z solvables: 422
rpm-md repo 'hardware_razer' (cached); generated: 2024-05-13T15:52:54Z solvables: 15
rpm-md repo 'updates-archive' (cached); generated: 2024-05-22T01:41:39Z solvables: 13161
Resolving dependencies... done
Checking out packages... done
Running pre scripts... done
Running post scripts... done
Running posttrans scripts... done
error: Executing %transfiletriggerin for glibc-common: bwrap(/usr/bin/sh): Child process killed by signal 7; run `journalctl -t 'rpm-ostree(glibc-common.transfiletriggerin)'` for more information
System update failed: 
   0: Command failed: `/usr/bin/rpm-ostree upgrade`
   1: `/usr/bin/rpm-ostree` failed: exit status: 1

Location:
   src/steps/os/linux.rs:273
Retry? (y)es/(N)o/(s)hell/(q)uiterror: Recipe `update` failed with exit code 1
ccomer@main-bazzite:~$ journalctl -t 'rpm-ostree(glibc-common.transfiletriggerin)'
Jun 13 19:46:11 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[13924]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
Jun 14 20:51:24 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[34099]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
Jun 15 00:06:25 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[82815]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
Jun 15 00:07:24 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[84511]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
Jun 15 00:08:16 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[86204]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
-- Boot 8034fe1075f4431495668cbdb481a7d6 --
Jun 18 19:53:56 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[27397]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
Jun 18 20:26:06 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[40129]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
Jun 18 20:27:26 main-bazzite rpm-ostree(glibc-common.transfiletriggerin)[41866]: /proc/self/fd/5: line 1:     3 Bus error               (core dumped) /sbin/ldconfig
```

**Comment by cgcomer** [comment]
I had some time to dig into this some tonight. Did some research that pointed me to using ostree fsck to identify errors in certain commits. So what follows is my results thus far:

TLDR version:
ostree fsck  **Result:** Shit ton of errors
rpm-ostree upgrade --check   Result: error: Loading sack: Updating rpm-md repo 'updates-archive': etc...
rpm-ostree cleanup -m
rpm-ostree upgrade --check   **Result:** Completed without errors. No updates available (expected)
ostree fsck  **Result:** Completed with no ugly red errors BUT "Segmentation fault (core dumped)"

I'm struggling to figure out how to proceed from here.  The results I get when searching seem to refer to this issue when someone is writing software, not running a terminal command.

Complete "transcript" below:

```
ccomer@main-bazzite:~$ ostree fsck
Validating refs...
Validating refs in collections...
Enumerating commits...
Verifying content integrity of 574 commit objects...
fsck objects (1/260418) [             ]   0%
error: In commits 45ad5ad941d39c21f16ead2536e540fe2ce86782aa5d9c2b96376ca849c96305, e689ef9d6f2061415c20514888fa9963a43049bd9fc82032d70f6569690fb376, 38503935bd5d1d3208c4e7f44ca90116284b5349dce34575318adef65b216820, 7977bc50d6198dddc8c34bcfbfbffaee9ffccaa61cec2f6265a348f8fbcb6509, 6de987b3364ee7e759ef0d3e3fe00a64a655be9beb4d344f08c86cc817f1c24f, 42aef9a2d31ef77eb5edda10b7881ed6d8ca644202dc990b238222d59101c199, a10ac03e9be7ff6b18d4e54d858bb88689fa5890ca7bb445b41998591a38f4fa, 0c60c17d6f905e2031f48ac21e11666ae6fa7ac83cd9c618dc64e034b6309833, ac1463923bdf22eb2449890a5cb79191c4a061336417448f2f0cddaa1205361a, c484defc2f9e9e3cb4e096190018702567598b011cdbf894b714f6f0360d8d1d, 2d4c38ca32c19ce56a519deae79fee4589f206dceaf7acfe0c793d1aa463d315, fe331e1220cc5496d96010d193b12d4397928afd15b047c7f259edc2e36c632d, 651481a05f0cab1e0c80b3494f8caedf17bedf4c8e5f060ff5467b192fbd6c1f, b153c71b8844418bc11d11654db5d875f5ff37cdbfde818aec668a376802e8e1, 5f41248220b44db9e2e4a35cb8c45429119ce2a3348a59246cb14e2dd5d17108, 9000ebd6c916dccdb0256339ed8becce11161ca50638a5860ca8500b135dba7f, e568cf8c479f6f3a44c1a09f875be0bb3143b7f40be144c2fd50ab26af83445d, 919dc838b41f3fef158e80e6b7eee292bd425f824528918ba98172320e2a4ba9, 1baffac2ba94486d9ac44414df31867fcfc9d1528a9356178e59aa57ff4690f5: fsck content object 50c3e9ce1e2a71ef60953f7e22c428cacecbb34a017ab418f736faa3671db2c9: Opening content object 50c3e9ce1e2a71ef60953f7e22c428cacecbb34a017ab418f736faa3671db2c9: openat: Permission denied
ccomer@main-bazzite:~$ rpm-ostree upgrade --check
Enabled rpm-md repositories: copr:copr.fedorainfracloud.org:matte-schwartz:sunshine copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager copr:copr.fedorainfracloud.org:rodoma92:rmlint copr:copr.fedorainfracloud.org:rok:cdemu fedora-cisco-openh264 updates fedora rpmfusion-free-updates-testing rpmfusion-free-updates rpmfusion-free hardware_razer updates-archive
Updating metadata for 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:rmlint'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rok:cdemu'... done
Updating metadata for 'fedora-cisco-openh264'... done
Updating metadata for 'updates'... done
Updating metadata for 'fedora'... done
Updating metadata for 'rpmfusion-free-updates-testing'... done
Updating metadata for 'rpmfusion-free-updates'... done
Updating metadata for 'rpmfusion-free'... done
Updating metadata for 'hardware_razer'... done
Updating metadata for 'updates-archive'... done
error: Loading sack: Updating rpm-md repo 'updates-archive': repodata updates-archive was not complete: Checksum error /var/cache/rpm-ostree/repomd/updates-archive-40-x86_64/repodata/966003c25a3507b5bcfa4caf567198eb074078daecd7d0afdff11e931eae25ea-primary.xml.zck: Unable to validate zchunk checksums
ccomer@main-bazzite:~$ rpm-ostree cleanup -m
ccomer@main-bazzite:~$ rpm-ostree upgrade --check
Enabled rpm-md repositories: copr:copr.fedorainfracloud.org:matte-schwartz:sunshine copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager copr:copr.fedorainfracloud.org:rodoma92:rmlint copr:copr.fedorainfracloud.org:rok:cdemu fedora-cisco-openh264 updates fedora rpmfusion-free-updates-testing rpmfusion-free-updates rpmfusion-free hardware_razer updates-archive
Updating metadata for 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rodoma92:rmlint'... done
Updating metadata for 'copr:copr.fedorainfracloud.org:rok:cdemu'... done
Updating metadata for 'fedora-cisco-openh264'... done
Updating metadata for 'updates'... done
Updating metadata for 'fedora'... done
Updating metadata for 'rpmfusion-free-updates-testing'... done
Updating metadata for 'rpmfusion-free-updates'... done
Updating metadata for 'rpmfusion-free'... done
Updating metadata for 'hardware_razer'... done
Updating metadata for 'updates-archive'... done
Importing rpm-md... done
rpm-md repo 'copr:copr.fedorainfracloud.org:matte-schwartz:sunshine'; generated: 2024-06-02T08:36:27Z solvables: 2
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:kde-cdemu-manager'; generated: 2024-04-20T12:39:00Z solvables: 12
rpm-md repo 'copr:copr.fedorainfracloud.org:rodoma92:rmlint'; generated: 2024-05-10T10:16:28Z solvables: 4
rpm-md repo 'copr:copr.fedorainfracloud.org:rok:cdemu'; generated: 2024-02-20T12:07:53Z solvables: 23
rpm-md repo 'fedora-cisco-openh264'; generated: 2024-03-12T11:45:42Z solvables: 3
rpm-md repo 'updates'; generated: 2024-06-20T01:46:01Z solvables: 18433
rpm-md repo 'fedora'; generated: 2024-04-14T18:51:11Z solvables: 74881
rpm-md repo 'rpmfusion-free-updates-testing'; generated: 2024-06-14T06:38:19Z solvables: 13
rpm-md repo 'rpmfusion-free-updates'; generated: 2024-06-14T06:37:04Z solvables: 127
rpm-md repo 'rpmfusion-free'; generated: 2024-04-20T12:11:51Z solvables: 422
rpm-md repo 'hardware_razer'; generated: 2024-05-13T15:52:54Z solvables: 15
rpm-md repo 'updates-archive'; generated: 2024-05-22T01:41:39Z solvables: 13161
Note: --check and --preview may be unreliable.  See https://github.com/coreos/rpm-ostree/issues/1579
No updates available.
ccomer@main-bazzite:~$ ostree fsck
Validating refs...
Validating refs in collections...
Enumerating commits...
Verifying content integrity of 574 commit objects...
Segmentation fault (core dumped)
```

**Comment by cgcomer** [comment]
Just an update.  See also #1326. First, I have not attempted a manual System update since hitting a wall with this, and have stepped back to see if Bazzite's auto update will manage things properly.

And it appears to be going well.  I have not had an issue with Flatpaks not updating and Bazzite a couple of weeks ago (after the manual update issue) appears to have been keeping itself up to date.

```
rpm-ostree status
State: idle
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
                   Digest: sha256:595da95b9a6fb6fe24839e04e7510f115997474b5591711d25fbdc8f55086995
                  Version: 40.20240803.0 (2024-08-03T15:34:43Z)
          LayeredPackages: edk2-ovmf qemu virt-manager
            LocalPackages: lact-0.5.4-0.x86_64
                Initramfs: regenerate

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite:stable
                   Digest: sha256:3afd6dbff34fa7d118d30c4186ec181c0d80ac64d607e09e6923ae6c7d05cd3c
                  Version: 40.20240802.0 (2024-08-02T16:17:11Z)
          LayeredPackages: edk2-ovmf qemu virt-manager
            LocalPackages: lact-0.5.4-0.x86_64
                Initramfs: regenerate
```

Like I said on the other Issue, I don't know what the problem was specifically but it appears to resolved itself somehow.

**Comment by HikariKnight** [comment]
At least youre updated and it works now, i am closing the issues but if any of these happen again open a new issue :)
